src/data/bucket/line_bucket.js

```diff
@@ -357,7 +358,21 @@ class LineBucket implements Bucket {
         }
 
         // Ignore invalid geometry.
-        if (len < (isPolygon ? 3 : 2)) return;
+        const count = len - first;
+        if (count < (isPolygon ? 3 : 2)) {
+            if (count === 0 || !collapse) {
+                return;
+            }
+            //Allow lines/polygons to collapse to points
+            if (count === 1) {
+                if (len < vertices.length) {
+                    len++;
+                } else if (first > 0) {
+                    first--;
+                }
```

Comment:
Oh, hold on……… I think I accomplished this in a slightly different but actually exactly equivalent way. 👍 👍  I need to process the polygon part a bit, but this block looks great to me 👍